test/abstract_syntax_tree_location_data.coffee

```diff
@@ -2438,3 +2438,249 @@ test "AST location Data as expected for CSXTag node", ->
         line: 1
         column: 31
 
+test "AST as expected for Try node", ->
+  testAstLocationData 'try cappuccino',
+    type: 'TryStatement'
+    block:
+      type: 'BlockStatement'
+      body: [
+        expression:
+          start: 4
+          end: 14
+          range: [4, 14]
+          loc:
+            start:
+              line: 1
+              column: 4
+            end:
+              line: 1
+              column: 14
+        start: 4
+        end: 14
+        range: [4, 14]
+        loc:
+          start:
+            line: 1
+            column: 4
+          end:
+            line: 1
+            column: 14
+      ]
+      # TODO: the location data for the block is wrong (wrong location data on generated INDENT)
```

Comment:
This gives a taste of the kind of generated `Block`/`INDENT`/`OUTDENT`/comments/location data stuff we'll run into. Left this particular case as a TODO as we can deal with this stuff in its own later PR

Reply:
Everything here looks really good, this is my only hesitation. Why is this location data incorrect? Is it something particular to `try` or rather to all blocks?

Reply:
It applies to blocks in general - basically in the rewriter (eg `fixOutdentLocationData()`) there's some stuff going on with assigning location data for generated tokens like `INDENT`/`OUTDENT` that isn't accurate enough for our AST purposes

Now that we're getting into blocks, I could do a PR soon for this - I do think it'll be cleaner to do in its own PR since it applies to blocks in general and (like most things in the rewriter) takes a bit of mental energy to think through

Reply:
Okay I’ll merge this in, and then do you mind doing the blocks PR next? I feel like we should fix that sooner than later, before we do a bunch of other PRs for the other indented node classes like `If` and so on.

Reply:
@GeoffreyBooth sure that makes sense